### ISSUE TITLE
Build angular library with bazel 

### ISSUE THREAD

**Issue by dhondiramjawade** [bazelbuild/rules_nodejs#1366]
# 🐞 bug report
Getting an error while integrating Bazel in an angular library.
### Affected Rule

 The issue is caused by the rule: 

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

### Description

I want to use Bazel with the angular library but getting some issues. I create a BUILD.bazel file in projects/lib1/src folder and write a ng_module rule in bazel file, then I build target by running <code>bazel build //projects/lib1/src:lib1 </code>  command (here lib1 is library target name) it is working fine and generate all stuff in  dist/bin/projects/lib1/src/lib/ folder. 
but when I import lib1 in my application and add an entry of lib target into //src target  as follow
ng_module(
   name = "src",
   srcs = //something
   deps= [
     "//projects/lib1/src:lib1"
    .....
  ]
and run <code> bazel run //src:devserver </code> then I am getting an error like 
src/app/app.module.ts(7,28): error TS2307: Cannot find module 'lib1'.

i have uploaded my application on git you can check here 
https://github.com/dhondiramjawade/bzl-a-lib

 A clear and concise description of the problem...

## 🔥 Exception or Error

<pre> ng serve <code>

INFO: Analyzed target //src:devserver (363 packages loaded, 16165 targets configured).
INFO: Found 1 target...
[0 / 11] [Prepa] BazelWorkspaceStatusAction stable-status.txt
[28 / 29] Compiling Angular templates (ngc - devmode) //src:src; 1s worker
ERROR: /home/ist/innoeye/bazel/temp1/src/BUILD.bazel:24:1: Compiling Angular templates (ngc - devmode) //src:src failed (Exit 1)
src/app/app.module.ts(7,28): error TS2307: Cannot find module 'lib1'.

Target //src:devserver failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /home/ist/innoeye/bazel/temp1/src/BUILD.bazel:88:1 Compiling Angular templates (ngc - devmode) //src:src failed (Exit 1)
INFO: Elapsed time: 7.139s, Critical Path: 4.29s
INFO: 2 processes: 2 linux-sandbox.
FAILED: Build did NOT complete successfully
ERROR: Build failed. Not running target
FAILED: Build did NOT complete successfully
Error starting process: fork/exec /tmp/bazel_script_path650036313: permission denied
Run start failed fork/exec /tmp/bazel_script_path650036313: permission denied
Triggering live reload
requesting reload: reload

</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>

  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>

  </code>
</pre>

**Rules version (SHA):**

<pre>
  <code>

  </code>
</pre>

**Anything else relevant?**

**Comment by alexeagle** [comment]
This report is incomplete and a bit hard to follow. My guess is that you `import * from 'lib1'` but didn't tell Bazel to create a path-mapping in the typescript config that says where `lib1` comes from, so you should use the `module_name` attribute on the rule that declares `lib1`

**Comment by dhondiramjawade** [comment]
@alexeagle sorry for the incomplete report, I am new on GitHub but I am trying my best.
 As u suggested and I already tried this by addeding <code> module_name </code> in libaray.
 It look likes 
<pre>
demo/projects/lib1/src/BUILD.bazel

<code>ng_module(
    name = "lib1",
    <b>module_name = "lib1",</b>
    srcs = glob(
        include = ["*.ts","**/*.ts"],
        exclude = [
            "**/*.spec.ts",
            "test.ts"
        ],
    ),.......</code></pre>

<pre>
demo/src/BUILD.bazel
<code>
ng_module(
    name = "src",
    srcs = glob(
        include = ["**/*.ts"],
        exclude = [
            "**/*.spec.ts",
            "main.ts",
            "test.ts",
            "initialize_testbed.ts",
        ],
    ),
    assets = glob([
      "**/*.css",
      "**/*.html",
    ]) + ([":styles"] if len(glob(["**/*.scss"])) else []),
    deps = [
     <b>   "//projects/lib1/src:lib1", </b>
.............................
</code></pre>
and import lib module in app module
<pre>
demo/src/app/app.module.ts
<code> 
import { Lib1Module } from "lib1";
</code></pre>



but still I am facing same issues.
<pre><code>
src/app/app.module.ts(7,28): error TS2307: <b>Cannot find module 'lib1'.</b>
</code></pre>

I tried to add the path in <b>ts.config.json </b> but it doesn't work for me. 
Is this a correct way to integrate the library with Bazel or should I have to use another rule for this?
like <b>ts_library , nodejs_binary ... </b>

**Comment by dhondiramjawade** [comment]
@alexeagle **I have resolved my issue.  thanks for your support.**

If anyone have same problem then check my repo i.e [https://github.com/dhondiramjawade/bzl-a-lib](url)

To resolve this issue i used **ts_library** rule which I already tried, but now I renamed public.api.ts with index.ts and added bazel file with exported members of index.ts. And finally added exported members bazel file to the **ts_library** rule as a dependency. 

As @alexeagle explained use `module_name` attribute to the `ts_library`. It will be helpful to import this library to any other module with module name, like` import ( SomeThing ) From "@yash/lib"`

<pre>ts_library(
    name = "lib",
    srcs = ["index.ts"],
    module_name = "@yash/lib",
    deps = [
        "//projects/lib1/src/lib:lib1"
    ]
)</pre>

for more information u can check my repo.

I am not sure but **ts_library** has some problem with public.api.ts file name, but  when i renamed it with index.ts then it worked for me.

**Comment by dhondiramjawade** [comment]
**Finally closing this ...**
